apis/server/router.go

```diff
@@ -24,54 +24,53 @@ func initRoute(s *Server) http.Handler {
 	r := mux.NewRouter()
 
 	// system
-	addRoute(r, http.MethodGet, "/_ping", s.ping)
-	addRoute(r, http.MethodGet, "/info", s.info)
-	addRoute(r, http.MethodGet, "/version", s.version)
-	addRoute(r, http.MethodPost, "/auth", s.auth)
+	s.addRoute(r, http.MethodGet, "/_ping", s.ping)
+	s.addRoute(r, http.MethodGet, "/info", s.info)
+	s.addRoute(r, http.MethodGet, "/version", s.version)
+	s.addRoute(r, http.MethodPost, "/auth", s.auth)
 
 	// daemon, we still list this API into system manager.
-	addRoute(r, http.MethodPost, "/daemon/update", s.updateDaemon)
+	s.addRoute(r, http.MethodPost, "/daemon/update", s.updateDaemon)
 
 	// container
-	addRoute(r, http.MethodPost, "/containers/create", s.createContainer)
-	addRoute(r, http.MethodPost, "/containers/{name:.*}/start", s.startContainer)
-	addRoute(r, http.MethodPost, "/containers/{name:.*}/stop", s.stopContainer)
-	addRoute(r, http.MethodPost, "/containers/{name:.*}/attach", s.attachContainer)
-	addRoute(r, http.MethodGet, "/containers/json", s.getContainers)
-	addRoute(r, http.MethodGet, "/containers/{name:.*}/json", s.getContainer)
-	addRoute(r, http.MethodDelete, "/containers/{name:.*}", s.removeContainers)
-	addRoute(r, http.MethodPost, "/containers/{name:.*}/exec", s.createContainerExec)
-	addRoute(r, http.MethodPost, "/exec/{name:.*}/start", s.startContainerExec)
-	addRoute(r, http.MethodGet, "/exec/{name:.*}/json", s.getExecInfo)
-	addRoute(r, http.MethodPost, "/containers/{name:.*}/rename", s.renameContainer)
-	addRoute(r, http.MethodPost, "/containers/{name:.*}/restart", s.restartContainer)
-	addRoute(r, http.MethodPost, "/containers/{name:.*}/pause", s.pauseContainer)
-	addRoute(r, http.MethodPost, "/containers/{name:.*}/unpause", s.unpauseContainer)
-	addRoute(r, http.MethodPost, "/containers/{name:.*}/update", s.updateContainer)
-	addRoute(r, http.MethodPost, "/containers/{name:.*}/upgrade", s.upgradeContainer)
-	addRoute(r, http.MethodGet, "/containers/{name:.*}/top", s.topContainer)
-	addRoute(r, http.MethodGet, "/containers/{name:.*}/logs", s.logsContainer)
-	addRoute(r, http.MethodPost, "/containers/{name:.*}/resize", s.resizeContainer)
-	addRoute(r, http.MethodPost, "/containers/{name:.*}/restart", s.restartContainer)
+	s.addRoute(r, http.MethodPost, "/containers/create", s.createContainer)
+	s.addRoute(r, http.MethodPost, "/containers/{name:.*}/start", s.startContainer)
+	s.addRoute(r, http.MethodPost, "/containers/{name:.*}/stop", s.stopContainer)
+	s.addRoute(r, http.MethodPost, "/containers/{name:.*}/attach", s.attachContainer)
+	s.addRoute(r, http.MethodGet, "/containers/json", s.getContainers)
+	s.addRoute(r, http.MethodGet, "/containers/{name:.*}/json", s.getContainer)
+	s.addRoute(r, http.MethodDelete, "/containers/{name:.*}", s.removeContainers)
+	s.addRoute(r, http.MethodPost, "/containers/{name:.*}/exec", s.createContainerExec)
+	s.addRoute(r, http.MethodPost, "/exec/{name:.*}/start", s.startContainerExec)
```

Comment:
I think it is your missing line of code `addRoute(r, http.MethodGet, "/exec/{name:.*}/json", s.getExecInfo)` leads to the CI failure.